src/hexdoc/_templates/components/table_of_contents.html.jinja

```diff
@@ -26,3 +26,18 @@
     </details>
   {% endfor %}
 </nav>
+
+<script>
+// Add this script right before the closing body tag or in a separate JS file
+document.addEventListener('DOMContentLoaded', (event) => {
+  const btn = document.getElementById('collapse-button');
+  const tocContainer = document.querySelector('.toc-container');
+
+  btn.addEventListener('click', function() {
+    // Toggle the ToC container visibility
+    tocContainer.style.display = tocContainer.style.display === 'none' ? 'block' : 'none';
+    // Update button text accordingly
+    btn.textContent = tocContainer.style.display === 'none' ? 'Expand' : 'Collapse';
```

Comment:
These should be localized. Take a look at the expand/collapse button for crafting and patterns.